table.go

```diff
@@ -1186,6 +1186,26 @@ func (t *TableBlock) Insert(ctx context.Context, tx uint64, buf *dynparquet.Seri
 	return nil
 }
 
+func (t *TableBlock) replaceIndex(replace func(index *btree.BTree) error) error {
```

Comment:
Could you add a small comment as to how this should be used? On the first read it was unclear if the callback was the one doing the replacing. Since caller's don't need to know the implementation details of how an index is updated, maybe this makes more sense as `updateIndex`?:
```suggestion
// updateIndex updates the TableBlock's index to the result of calling updateFn with the
// current index.
func (t *TableBlock) updateIndex(updateFn func(index *btree.BTree) error) error {
```